src/ARPlanes.js

```diff
@@ -63,108 +63,122 @@ class ARPlanes extends Object3D {
   constructor(vrDisplay) {
     super();
     this.vrDisplay = vrDisplay;
-    this.planes = [];
+    this.planes = new Map();
 
     // A mapping of plane IDs to colors, so that we can reuse the same
     // color everytime we update for the same plane rather than randomizing
     // @TODO When we have plane removal events, clear this map so we don't
     // have a leak
     this.materialMap = new Map();
-  }
-
-  /**
-   * Clear out the THREE representation mesh from
-   * scene.
-   */
-  clear() {
-    this.planes.forEach(plane => this.remove(plane));
-    this.planes.length = 0;
-  }
 
-  /**
-   * Polling callback while enabled, used to fetch and orchestrate
-   * plane rendering. If successful, returns the number of planes found.
-   *
-   * @return {number?}
-   */
-  update() {
-    if (!this.vrDisplay || !this.vrDisplay.getPlanes) {
-      return;
+    const addPlane = (plane) => {
+      let planeObj = this.createPlane(plane);
+      if (planeObj) {
+        this.add(planeObj);
+        this.planes.set(plane.identifier, planeObj);
+      }
     }
 
-    // Remove current planes and clear out
-    // from scene
-    this.clear();
+    const removePlane = (identifier) => {
+      let existing = this.planes.get(identifier);
+      if (existing) {
+        this.remove(existing);
+      }
+      this.planes.delete(identifier);
+    }
+  
+    vrDisplay.addEventListener('planesadded', (event) => {
+      console.log(event.planes);
+      event.planes.forEach(addPlane);
+    });
+
+    vrDisplay.addEventListener('planesupdated', (event) => {
+      for (let plane of event.planes) {
+        removePlane(plane.identifier);
+        addPlane(plane);
+      }
+    });
 
-    // Recreate each plane detected
-    const planes = this.vrDisplay.getPlanes();
-    for (const anchor of planes) {
-      if (anchor.vertices.length == 0) {
-        continue;
+    vrDisplay.addEventListener('planesremoved', (event) => {
+      for (let plane of event.planes) {
+        removePlane(plane.identifier);
       }
+    });
+  }
 
-      const id = anchor.identifier;
-      const planeObj = new Object3D();
-      const mm = anchor.modelMatrix;
-      planeObj.matrixAutoUpdate = false;
-      planeObj.matrix.set(
-        mm[0],
-        mm[4],
-        mm[8],
-        mm[12],
-        mm[1],
-        mm[5],
-        mm[9],
-        mm[13],
-        mm[2],
-        mm[6],
-        mm[10],
-        mm[14],
-        mm[3],
-        mm[7],
-        mm[11],
-        mm[15]
+  createPlane(plane) {
+    if (plane.vertices.length == 0) {
+      return null;
+    }
+
+    const id = plane.identifier;
+    const planeObj = new Object3D();
```

Comment:
Is there a purpose of having the Plane Mesh be wrapped by an Object3D? Can we just use the Mesh?

Reply:
done